clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/DaVinciRecordTransformerUtility.java

```diff
@@ -116,7 +122,7 @@ public final void onRecovery(
       for (iterator.seekToFirst(); iterator.isValid(); iterator.next()) {
         byte[] keyBytes = iterator.key();
         byte[] valueBytes = iterator.value();
-        Lazy<K> lazyKey = Lazy.of(() -> getKeyDeserializer().deserialize(ByteBuffer.wrap(keyBytes)));
+        Lazy<K> lazyKey = Lazy.of(() -> keyDeserializer.deserialize(ByteBuffer.wrap(keyBytes)));
```

Comment:
Nitpick: There is no need to wrap into a `ByteBuffer`. There is also an API which takes in a `byte[]` directly. The `BB` can be useful when we need to use just a subset of the bytes, as is the case for the value, below.